packages/vite/src/node/server/environments/nodeEnvironment.ts

```diff
@@ -1,15 +1,27 @@
 import type { ResolvedConfig } from '../../config'
 import type { DevEnvironmentSetup } from '../environment'
 import { DevEnvironment } from '../environment'
+import { createServerHotChannel } from '../hmr'
 import { asyncFunctionDeclarationPaddingLineCount } from '../../../shared/utils'
 
+export function createNodeSsrDevEnvironment(
+  name: string,
+  config: ResolvedConfig,
+  setup?: DevEnvironmentSetup,
+): DevEnvironment {
+  return createNodeDevEnvironment(name, config, {
+    ...setup,
+    hot: createServerHotChannel(),
+  })
+}
```

Comment:
@sheremet-va do we need a separate `createNodeDevEnvironment()` here? Or should this two functions be merged?

Reply:
I am confused why it's needed. Just to have `(environmentName === 'ssr'
         ? createNodeSsrDevEnvironment
         : createDevEnvironment)`?

Maybe just make an environment creation a separate function without ternaries?

Reply:
I can move `createServerHotChannel()` to the config file, not problem there.

Reply:
Moved